src/sdk/k4a.c

```diff
@@ -526,171 +526,252 @@ static k4a_result_t validate_configuration(k4a_context_t *device, const k4a_devi
 {
     RETURN_VALUE_IF_ARG(K4A_RESULT_FAILED, config == NULL);
     RETURN_VALUE_IF_ARG(K4A_RESULT_FAILED, device == NULL);
-    k4a_result_t result;
+    k4a_result_t result = K4A_RESULT_SUCCEEDED;
     bool depth_enabled = false;
     bool color_enabled = false;
 
-    result = K4A_RESULT_FROM_BOOL(
```

Comment:
>K4A_RESULT_FROM_BOOL [](start = 13, length = 20)

It looks good in general for the explicit logging part, but you might want to confirm with Wes about not using the K4A_RESULT_FROM_BOOL (which is not only return a k4a_result_t, but also does a TraceError). Theoretically, you can add explicit logging based on the result value without losing the trace logging as before too :)

Reply:
The feedback we are getting is that tracing from the macro is confusing, so for spots in the code that we think users will commonly encounter errors, like the validation of the config structure, we a providing clearer error messages. So I am good with this, thanks @rabbitdaxi.

Reply:
Good to know :)

---
In reply to: [304560357](https://github.com/microsoft/Azure-Kinect-Sensor-SDK/pull/513#discussion_r304560357) [](ancestors = 304560357)